### PR TITLE
Booleans and comparison ops

### PR DESCRIPTION
**Pull request by wezm**
This PR adds:

- Booleans
- Boolean operators
- Comparison operators

### CODE REVIEW

**Comment by mikeday** [comment]
Can we and a u16 with another and check if the result is non-zero now? :nerd_face:

**Comment by wezm** [comment]
> Can we and a u16 with another and check if the result is non-zero now? 🤓

Yes I think so: 

```
$ echo 'u16_neq (u16_add 1 2) 0' | cr norm
true : Bool
```

**Review by brendanzab — APPROVED** [review]
Wonderful 👍

**Comment by wezm** [comment]
I noticed something else that I should include in this PR: an update to the reference. I'll add that at some point today.

**Comment by wezm** [comment]
Now includes update to the reference.

File: doc/reference.md [hunk 1]
```diff
@@ -567,6 +569,21 @@ some-data.data : Array3 some-data.len S32
 //                                      in type of the projected term
 ```
 
+## Booleans
+
+The boolean type is: `Bool` with values `true` and `false`.
```
Comment (brendanzab) [inline doc/reference.md:574]: I must admit I never really know when to use the term ‘value’ now, but I think this is ok given that these terms evaluate to these values? Might be something to ask type theory twitter about heh 🤔

**Review by brendanzab — APPROVED** [review]
Looks good!

**Pull request merged**